src/scripts/components/VariableMessage.jsx

```diff
@@ -18,24 +20,43 @@ import {
 } from '../constants';
 
 class VariableMessage extends React.Component {
+  currentDraggedSpanVariable = '';
+  isCategoryMobileViewVisible = false;
   state = {
     message: '',
     available: [],
+    variables: [],
+    variablesOfCategory: [],
+    categories: [],
+    selectedCategory: '',
   };
 
   componentWillMount() {
+    const toUpdateState = {
+      categories: [],
+    };
+
+    if (this.props.isCategoryAvailable) {
+      this.isCategoryMobileViewVisible = window.outerWidth <= 767; //  for mobile view
+      toUpdateState.selectedCategory = this.props.defaultSelectedCategory;
+      toUpdateState.variablesOfCategory =
+        this.props.data.filter((item) => {
+          if (!toUpdateState.categories.includes(item.category)) {
+            toUpdateState.categories.push(item.category);
+          }
+          return (item.category === this.props.defaultSelectedCategory);
+        });
+    }
     if (this.props.initialValue) {
-      this.setState({
-        message: this.props.initialValue,
-      });
+      toUpdateState.message = this.props.initialValue;
     }
+    toUpdateState.variables = this.props.data;
+    this.setState(toUpdateState);
```

Comment:
In general if all we want to do with `componentWillMount` is to set initial state based on props, then we can do that in the state declaration.

E.G:
```
state = {
  selectedCategory = this.props.selectedCategory || '',
  ...etc
}
```

Moreover if these items are not changing at all, then it would behoove us to pass them as props and just use them, rather than go through the process of initializing derived state. Derived state needs to then be managed everyime props change so it has the potential to become cumbersome for maintainability. At the moment it seems we're ONLY reacting to the `initialValue` prop change but the pattern is not scalable.